### ISSUE TITLE
netdata fails to parse DIMENSIONS in SNMP collector config, causing node.js plugin to crash

### ISSUE THREAD

**Issue by nega0** [netdata/netdata#4075]
### tl;dr
Whitespace in DIMENSION ids causes `node.d.plugin` to crash (exit with error code -1).

### Details
<details><summary>My <tt>/opt/netdata/etc/netdata/node.d/snmp.conf:</tt></summary>
<pre> 
{
    "enable_autodetect": false,
        "update_every": 15,
        "servers": [
        {
            "hostname": "172.16.172.79",
            "community": "public",
            "update_every": 15,
            "charts": {
                "snmp_netbotz250.temperature_probe1": {
                    "title": "NetBotz250 Temperature Probe 1",
                    "units": "Celsius",
                    "type": "line",
                    "family": "tempz",
                    "dimensions": {
                        "Probe 1": {
                            "oid": ".1.3.6.1.4.1.318.1.1.10.4.2.3.1.5.0.1",
                            "offset": 0
                        },
                        "Probe 2": {
                            "oid": ".1.3.6.1.4.1.318.1.1.10.4.2.3.1.5.0.2",
                            "offset": 0
                        },
                        "Probe 3": {
                            "oid": ".1.3.6.1.4.1.318.1.1.10.4.2.3.1.5.0.3",
                            "offset": 0
                        }
                    }
                }
            }
        }
    ]
}
</pre></details>

Note the DIMENSION ids "Probe 1" "Probe 2" and "Probe 3"

<details><summary>Running the plugin by itself, this config works as expected and collects data every 15 seconds.</summary><pre>
[netdata:~] sudo -u netdata /opt/netdata/usr/libexec/netdata/plugins.d/node.d.plugin 1
2018-08-23 15:54:48: node.d.plugin: ERROR: Cannot load module's stiebeleltron config from /opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/stiebeleltron.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/stiebeleltron.conf', using internal defaults.
2018-08-23 15:54:48: node.d.plugin: ERROR: Cannot load module's sma_webbox config from /opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/sma_webbox.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/sma_webbox.conf', using internal defaults.
2018-08-23 15:54:48: node.d.plugin: ERROR: Cannot load module's named config from /opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/named.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/named.conf', using internal defaults.
2018-08-23 15:54:48: node.d.plugin: ERROR: Cannot load module's fronius config from /opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/fronius.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/fronius.conf', using internal defaults.
2018-08-23 15:54:48: node.d.plugin: ERROR: named: local: Failed to make request, message: connect ECONNREFUSED 127.0.0.1:8888
CHART "snmp_netbotz250.temperature_probe1" "snmp_netbotz250.temperature_probe1" "NetBotz250 Temperature Probe 1" "Celsius" "tempz" "" "line" 50000 15
DIMENSION "Probe 1" "Probe 1" "absolute" 1 1 
DIMENSION "Probe 2" "Probe 2" "absolute" 1 1 
DIMENSION "Probe 3" "Probe 3" "absolute" 1 1 
BEGIN snmp_netbotz250.temperature_probe1 
SET Probe 1 = 18
SET Probe 2 = 14
SET Probe 3 = 19
END

BEGIN snmp_netbotz250.temperature_probe1 11309000
SET Probe 1 = 18
SET Probe 2 = 14
SET Probe 3 = 19
END

^C</pre></details>

<details><summary>Running this config as part of <tt>netdata</tt> causes errors, the <tt>node.d.plugin</tt> to "Exit with error code -1", and no updates to the graph.
</summary><pre>
[netdata:~] cat /dev/null | sudo tee /opt/netdata/var/log/netdata/error.log           
[netdata:~] sudo service netdata restart                                   
[netdata:~] !grep
grep -Ei "(snmp|node\.d\.plugin)" /opt/netdata/var/log/netdata/error.log
2018-08-23 15:56:26: netdata ERROR : PLUGINSD[node.d] : '/opt/netdata/usr/libexec/netdata/plugins.d/node.d.plugin' (pid 11599) disconnected after 87 successful data collections (ENDs).
2018-08-23 15:56:26: netdata ERROR : PLUGINSD[node.d] : '/opt/netdata/usr/libexec/netdata/plugins.d/node.d.plugin' (pid 11599) exited with error code -1, but has given useful output in the past (87 times). Waiting a bit before starting it again.
2018-08-23 15:56:27: netdata INFO  : PLUGINSD[node.d] : connected to '/opt/netdata/usr/libexec/netdata/plugins.d/node.d.plugin' running on pid 15173
2018-08-23 15:56:28: node.d.plugin: ERROR: Cannot load module's stiebeleltron config from /opt/netdata/etc/netdata/node.d/stiebeleltron.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/etc/netdata/node.d/stiebeleltron.conf', using internal defaults.
2018-08-23 15:56:28: node.d.plugin: ERROR: Cannot load module's sma_webbox config from /opt/netdata/etc/netdata/node.d/sma_webbox.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/etc/netdata/node.d/sma_webbox.conf', using internal defaults.
2018-08-23 15:56:28: node.d.plugin: ERROR: Cannot load module's named config from /opt/netdata/etc/netdata/node.d/named.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/etc/netdata/node.d/named.conf', using internal defaults.
2018-08-23 15:56:28: netdata ERROR : PLUGIN[proc] : PROCFILE: Cannot open file '/proc/net/snmp6' (errno 2, No such file or directory)
2018-08-23 15:56:28: netdata ERROR : PLUGIN[proc] : PROCFILE: Cannot open file '/proc/net/sctp/snmp' (errno 2, No such file or directory)
2018-08-23 15:56:28: node.d.plugin: ERROR: Cannot load module's fronius config from /opt/netdata/etc/netdata/node.d/fronius.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/etc/netdata/node.d/fronius.conf', using internal defaults.
2018-08-23 15:56:28: node.d.plugin: ERROR: named: local: Failed to make request, message: connect ECONNREFUSED 127.0.0.1:8888
2018-08-23 15:56:28: netdata ERROR : PLUGINSD[node.d] : requested a SET to dimension with id 'Probe' on stats 'snmp_netbotz250.temperature_probe1' (snmp_netbotz250.temperature_probe1) on host 'netdata', which does not exist. Disabling it.
2018-08-23 15:56:28: netdata ERROR : PLUGINSD[node.d] : '/opt/netdata/usr/libexec/netdata/plugins.d/node.d.plugin' (pid 15173) disconnected after 0 successful data collections (ENDs).
2018-08-23 15:56:28: netdata ERROR : PLUGINSD[node.d] : '/opt/netdata/usr/libexec/netdata/plugins.d/node.d.plugin' (pid 15173) exited with error code -1. Disabling it.
[netdata:~] </pre></details>

<br />
This line is the actual relevant error. 

```
2018-08-23 15:56:28: netdata ERROR : PLUGINSD[node.d] : requested a SET to dimension with id 'Probe' on stats 'snmp_netbotz250.temperature_probe1' (snmp_netbotz250.temperature_probe1) on host 'netdata', which does not exist. Disabling it.
```

Based on [External Plugins::Dimension](https://github.com/firehol/netdata/wiki/External-Plugins#dimension) this behavior was unexpected. 

<details><summary>Changing the DIMENSION ids to "Probe_1" "Probe_2" "Probe_3" causes both standalone execution of the plugin to work and, execution within <tt>netdata</tt> to work. 
</summary><pre>
[netdata:~] sudo -u netdata /opt/netdata/usr/libexec/netdata/plugins.d/node.d.plugin 1
2018-08-23 16:06:26: node.d.plugin: ERROR: Cannot load module's stiebeleltron config from /opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/stiebeleltron.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/stiebeleltron.conf', using internal defaults.
2018-08-23 16:06:26: node.d.plugin: ERROR: Cannot load module's sma_webbox config from /opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/sma_webbox.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/sma_webbox.conf', using internal defaults.
2018-08-23 16:06:26: node.d.plugin: ERROR: Cannot load module's named config from /opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/named.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/named.conf', using internal defaults.
2018-08-23 16:06:26: node.d.plugin: ERROR: Cannot load module's fronius config from /opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/fronius.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/fronius.conf', using internal defaults.
2018-08-23 16:06:26: node.d.plugin: ERROR: named: local: Failed to make request, message: connect ECONNREFUSED 127.0.0.1:8888
CHART "snmp_netbotz250.temperature_probe1" "snmp_netbotz250.temperature_probe1" "NetBotz250 Temperature Probe 1" "Celsius" "tempz" "" "line" 50000 15
DIMENSION "Probe_1" "Probe_1" "absolute" 1 1 
DIMENSION "Probe_2" "Probe_2" "absolute" 1 1 
DIMENSION "Probe_3" "Probe_3" "absolute" 1 1 
BEGIN snmp_netbotz250.temperature_probe1 
SET Probe_1 = 18
SET Probe_2 = 15
SET Probe_3 = 19
END

BEGIN snmp_netbotz250.temperature_probe1 3697000
SET Probe_1 = 18
SET Probe_2 = 15
SET Probe_3 = 19
END

^C
[netdata:~] cat /dev/null | sudo tee /opt/netdata/var/log/netdata/error.log           
[netdata:~] sudo service netdata restart                                              
[netdata:~] grep -Ei "(snmp|node\.d\.plugin)" /opt/netdata/var/log/netdata/error.log  
2018-08-23 16:09:05: netdata INFO  : PLUGINSD[node.d] : connected to '/opt/netdata/usr/libexec/netdata/plugins.d/node.d.plugin' running on pid 17530
2018-08-23 16:09:06: netdata ERROR : PLUGIN[proc] : PROCFILE: Cannot open file '/proc/net/snmp6' (errno 2, No such file or directory)
2018-08-23 16:09:06: netdata ERROR : PLUGIN[proc] : PROCFILE: Cannot open file '/proc/net/sctp/snmp' (errno 2, No such file or directory)
2018-08-23 16:09:06: node.d.plugin: ERROR: Cannot load module's stiebeleltron config from /opt/netdata/etc/netdata/node.d/stiebeleltron.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/etc/netdata/node.d/stiebeleltron.conf', using internal defaults.
2018-08-23 16:09:06: node.d.plugin: ERROR: Cannot load module's sma_webbox config from /opt/netdata/etc/netdata/node.d/sma_webbox.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/etc/netdata/node.d/sma_webbox.conf', using internal defaults.
2018-08-23 16:09:06: node.d.plugin: ERROR: Cannot load module's named config from /opt/netdata/etc/netdata/node.d/named.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/etc/netdata/node.d/named.conf', using internal defaults.
2018-08-23 16:09:06: node.d.plugin: ERROR: Cannot load module's fronius config from /opt/netdata/etc/netdata/node.d/fronius.conf exception: Error: ENOENT: no such file or directory, open '/opt/netdata/etc/netdata/node.d/fronius.conf', using internal defaults.
2018-08-23 16:09:06: node.d.plugin: ERROR: named: local: Failed to make request, message: connect ECONNREFUSED 127.0.0.1:8888
2018-08-23 16:09:06: netdata INFO  : PLUGINSD[node.d] : File /opt/netdata/var/cache/netdata/snmp_netbotz250.temperature_probe1/Probe_1.db is too old (last collected 1535054946 seconds ago, but the database is 59940 seconds). Clearing it.
2018-08-23 16:09:06: netdata INFO  : PLUGINSD[node.d] : File /opt/netdata/var/cache/netdata/snmp_netbotz250.temperature_probe1/Probe_2.db is too old (last collected 1535054946 seconds ago, but the database is 59940 seconds). Clearing it.
2018-08-23 16:09:06: netdata INFO  : PLUGINSD[node.d] : File /opt/netdata/var/cache/netdata/snmp_netbotz250.temperature_probe1/Probe_3.db is too old (last collected 1535054946 seconds ago, but the database is 59940 seconds). Clearing it.
</pre></details>
